jgrapht-core/src/main/java/org/jgrapht/alg/scoring/PageRank.java

```diff
@@ -0,0 +1,397 @@
+     */
+    public PageRank(Graph<V, E> g, double dampingFactor)
+    {
+        this(
+            g,
+            dampingFactor,
+            MAX_ITERATIONS_DEFAULT,
+            TOLERANCE_DEFAULT,
+            false);
+    }
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public PageRank(Graph<V, E> g, double dampingFactor, int maxIterations)
+    {
+        this(g, dampingFactor, maxIterations, TOLERANCE_DEFAULT, false);
+    }
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param maxIterations the maximum number of iterations to perform
+     * @param tolerance the calculation will stop if the difference of PageRank
+     *        values between iterations change less than this value
+     */
+    public PageRank(
+        Graph<V, E> g,
+        double dampingFactor,
+        int maxIterations,
+        double tolerance)
+    {
+        this(g, dampingFactor, maxIterations, tolerance, false);
+    }
+
+    /**
+     * Create and execute an instance of PageRank.
+     * 
+     * This constructor allows the caller to adjust all parameters as well as to
+     * use a weighted variant of the algorithm.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param maxIterations the maximum number of iterations to perform
+     * @param tolerance the calculation will stop if the difference of PageRank
+     *        values between iterations change less than this value
+     * @param weighted assume the edges are weighted or not. If {@code true} a
+     *        weighted variant is used where the probability of following an
+     *        edge e out of node v is equal to the weight of e over the sum of
+     *        weights of all outgoing edges of v.
+     */
+    public PageRank(
+        Graph<V, E> g,
+        double dampingFactor,
+        int maxIterations,
+        double tolerance,
+        boolean weighted)
+    {
+        this.g = g;
+        this.scores = new HashMap<>();
+        new AlgorithmDetails(
+            g,
+            dampingFactor,
+            maxIterations,
+            tolerance,
+            weighted).run();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map<V, Double> getScores()
+    {
+        return Collections.unmodifiableMap(scores);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Double getVertexScore(V v)
+    {
+        if (!g.containsVertex(v)) {
+            throw new IllegalArgumentException(
+                "Cannot return score of unknown vertex");
+        }
+        return scores.get(v);
+    }
+
+    /**
+     * The implementation details
+     */
+    class AlgorithmDetails
+    {
```

Comment:
I changed it to not use an inner class.

In more complicated algorithms where you have a lot of functions working on the same state, I find it quite helpful to keep state inside an inner class, instead of using local variables and passing around a lot of parameters.